servers/src/test/java/tachyon/worker/block/TieredBlockStoreTest.java

```diff
@@ -0,0 +1,267 @@
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.worker.block;
+
+import java.io.File;
+import java.lang.reflect.Field;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+import com.google.common.collect.Sets;
+
+import tachyon.StorageLevelAlias;
+import tachyon.conf.TachyonConf;
+import tachyon.exception.NotFoundException;
+import tachyon.exception.OutOfSpaceException;
+import tachyon.util.io.FileUtils;
+import tachyon.worker.block.evictor.Evictor;
+import tachyon.worker.block.meta.BlockMeta;
+import tachyon.worker.block.meta.StorageDir;
+import tachyon.worker.block.meta.TempBlockMeta;
+
+public class TieredBlockStoreTest {
+  private static final long USER_ID1 = 2;
+  private static final long USER_ID2 = 3;
+  private static final long BLOCK_ID1 = 1000;
+  private static final long BLOCK_ID2 = 1001;
+  private static final long TEMP_BLOCK_ID = 1003;
+  private static final long BLOCK_SIZE = 512;
+  private static final StorageLevelAlias FIRST_TIER_ALIAS = TieredBlockStoreTestUtils.TIER_ALIAS[0];
+
+  @Rule
+  public TemporaryFolder mTestFolder = new TemporaryFolder();
+  @Rule
+  public ExpectedException mThrown = ExpectedException.none();
+  private TieredBlockStore mBlockStore;
+  private BlockMetadataManager mMetaManager;
+  private BlockLockManager mLockManager;
+  private StorageDir mTestDir1;
+  private StorageDir mTestDir2;
+  private Evictor mEvictor;
+
+  @Before
+  public void before() throws Exception {
+    File tempFolder = mTestFolder.newFolder();
+    TachyonConf tachyonConf =
+        TieredBlockStoreTestUtils.defaultTachyonConf(tempFolder.getAbsolutePath());
+    mBlockStore = new TieredBlockStore(tachyonConf);
+
+    Field field = mBlockStore.getClass().getDeclaredField("mMetaManager");
+    field.setAccessible(true);
+    mMetaManager = (BlockMetadataManager) field.get(mBlockStore);
+    field = mBlockStore.getClass().getDeclaredField("mLockManager");
+    field.setAccessible(true);
+    mLockManager = (BlockLockManager) field.get(mBlockStore);
+    field = mBlockStore.getClass().getDeclaredField("mEvictor");
+    field.setAccessible(true);
+    mEvictor = (Evictor) field.get(mBlockStore);
```

Comment:
No it is not. The best practice is probably to create a class called `TieredBlockStoreForTest` to wrap around `TieredBlockStore` but expose those fields.